Add a guide for consuming previously ingested events

tupui opened the pull request:
Closes #599

This guide shows how you can consume events from a DB.

It uses Python with some very common/recommended libraries and I did not add any magic.

I still want to add a small paragraph on idempotency, but the gist should be here. Let me know if this is what you had in mind and if there is anything I should change.

Just FYI, next week I will be at a Python maintainer summit so might be slow to reply 😄 Thanks!

cc @briwylde08

briwylde08 commented:
Yay! This is so exciting! I'll have our reviewers take a look and we'll get back to you with any feedback. Thank you so much for putting this together.

tupui commented:
Oh Waou thanks for taking the time to fix the linter for me 🤩 There are not a lot of projects where maintainers take the time to help like that 👏

Locally I had issues to run the doc. I think now looking at the CI that it's because I had node 22 and need Node 18. I am not a frontend dev so I mostly poke around for stuff like these 😅

briwylde08 commented:
Of course! No worries at all, I can't help myself, that red dot just bothers me [😂](https://emojipedia.org/face-with-tears-of-joy)

briwylde08 commented:
Ok, great, thank you so much! I am giving the reviewers three weeks to review submissions as they come in, so it maybe be a minute before you hear a response (but I'll try to move it along quicker than that :))

tupui commented on docs/smart-contracts/guides/events/consume.mdx:
```diff
@@ -1,7 +1,251 @@
+
+## Setup
+
+In a virtual environment, install the Python dependencies:
```
Ok yes, I will add that 👍 I was not sure how deep I had to cover that part since any guide showing some Python would need that. i.e. in the end I think it would be good to have a general setup page for the guides. This way we can present setups for all the languages on the same page and then assume all is good in the following sections.

tupui commented on docs/smart-contracts/guides/events/consume.mdx:
````diff
@@ -1,7 +1,251 @@
+Base.metadata.create_all(engine)
+```
+
+By setting `echo=True` we can understand what is happening on the database. Creating the database table leads to the following logs:
````
I actually also never really use this and I should 😅 I typically rely on tests or a local DB and since here I did not want to introduce something else, I went with this.

tupui commented on docs/smart-contracts/guides/events/consume.mdx:
````diff
@@ -1,7 +1,251 @@
+```python
+import sqlalchemy
+
+with orm.Session(engine) as session:
````
I think the confusion here is that I introduce the imports when I need them. Here `orm` relies on the previous import at the top of the guide.

Maybe I should move all the imports at the top instead to make it clearer?

(For `sqlalchemy.orm` I need to double check the imports. One of these is a bit strange as it's a submodule IIRC.)

tupui reviewed (commented):
Thank you @ElliotFriend for the kind words.

For the prerequisite section, why not. In the end in both cases we have a MySQL DB so it should be the same. I will find some time to make the change 👍 (next week probably)

briwylde08 commented:
Hi! @tupui - just want to make sure we're on the same page here. We're waiting on you to move the ingesting part of the guide to the current guide before finalizing review, correct? (also, since you pretty much wrote two guides, we'll up the award to $500 worth of XLM)

tupui commented:
Thanks a lot @briwylde08 🤩 

I made an update to move the ingestion part. I still need to double check all the code since I moved a few things around and adjusted some parts a bit. I also put some Python code for the ingestion part so that this whole guide would have both JS and Python.

I hope to have more time to finish tomorrow.

tupui commented:
Oh, there is a conflict! I will fix this. Otherwise I did all the changes. Let me know how you find this 😃 I re-ran everything and regenerated all the outputs.

tupui commented:
Ok, resolved. Now it should be ready from my side 😄

briwylde08 commented:
Amazing, thank you! Yes, the docs are always a work in progress [😅](https://emojipedia.org/grinning-face-with-sweat) so fixing links is a constant. Appreciate this, @ElliotFriend will take a look soon!

ElliotFriend commented on docs/smart-contracts/guides/events/ingest.mdx:
```diff
@@ -26,18 +112,77 @@ model SorobanEvent {
 ## Query Events from Soroban RPC
 
-First, we'll need to query the events from Soroban RPC. This simple JavaScript example will use the `@stellar/stellar-sdk` library to make an RPC request using the `getEvents` method, filtering for all `transfer` events that are emitted by the native XLM contract.
+First, we'll need to query the events from Soroban RPC. This simple makes an RPC request using the `getEvents` method, filtering for all `transfer` events that are emitted by the native XLM contract.
```
```suggestion
First, we'll need to query the events from Soroban RPC. This simple example makes an RPC request using the `getEvents` method, filtering for all `transfer` events that are emitted by the native XLM contract.
```

ElliotFriend reviewed (commented):
Fantastic work, @tupui!! I left one small grammar fix, but once that's done, this is ready to go!

Thanks again! Really great job!

tupui commented:
Thank you @ElliotFriend 😊 Just added your fix.

ElliotFriend reviewed (approved):
Nice!! Thanks for getting this done!

briwylde08 commented:
This is fantastic @tupui! Thank you!

tupui commented:
Thank you both for the help and the review 😃 It was a really great experience!
